dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,567 @@
+            table_names = [
+                table_name
+                for table_name in self.db_client.table_names()
+                if table_name.startswith(prefix)
+            ]
+        else:
+            table_names = self.db_client.table_names()
+
+        for table_name in table_names:
+            self.db_client.drop_table(table_name)
+
+        self._delete_sentinel_table()
+
+    @lancedb_error
+    def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        if not self.is_storage_initialized():
+            self._create_sentinel_table()
+        elif truncate_tables:
+            for table_name in truncate_tables:
+                fq_table_name = self.make_qualified_table_name(table_name)
+                if not self.table_exists(fq_table_name):
+                    continue
+                self.db_client.drop_table(fq_table_name)
+                self.create_table(
+                    table_name=fq_table_name,
+                    schema=self.get_table_schema(fq_table_name),
+                )
+
+    @lancedb_error
+    def is_storage_initialized(self) -> bool:
+        return self.table_exists(self.sentinel_table)
+
+    def _create_sentinel_table(self) -> None:
+        """Create an empty table to indicate that the storage is initialized."""
+        self.create_table(schema=cast(LanceModel, NullSchema), table_name=self.sentinel_table)
+
+    def _delete_sentinel_table(self) -> None:
+        """Delete the sentinel table."""
+        self.db_client.drop_table(self.sentinel_table)
+
+    @lancedb_error
+    def update_stored_schema(
+        self,
+        only_tables: Iterable[str] = None,
+        expected_update: TSchemaTables = None,
+    ) -> Optional[TSchemaTables]:
+        super().update_stored_schema(only_tables, expected_update)
+        applied_update: TSchemaTables = {}
+
+        try:
+            schema_info = self.get_stored_schema_by_hash(self.schema.stored_version_hash)
+        except DestinationUndefinedEntity:
+            schema_info = None
+
+        if schema_info is None:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                "not found in the storage. upgrading"
+            )
+            self._execute_schema_update(only_tables)
+        else:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                f"inserted at {schema_info.inserted_at} found "
+                "in storage, no upgrade required"
+            )
+        return applied_update
+
+    def _execute_schema_update(self, only_tables: Iterable[str]) -> None:
+        for table_name in only_tables or self.schema.tables:
+            exists = self.table_exists(self.make_qualified_table_name(table_name))
+            if not exists:
+                self.create_table(
+                    self.make_qualified_table_name(table_name),
+                    schema=cast(LanceModel, NullSchema),
```

Comment:
will the actual schema then be inferred from the incoming data or how does this work?